lib/dm-migrations/adapters/dm-do-adapter.rb

```diff
@@ -196,9 +196,9 @@ def create_unique_index_statements(model)
         # @api private
         def property_schema_hash(property)
           dump_class = property.dump_class
-          type_map   = self.class.type_map
+          type_by_property_class = self.class.type_by_property_class(property.class)
 
-          schema = (type_map[property.class] || type_map[dump_class]).merge(:name => property.field)
+          schema = (type_by_property_class || self.class.type_map[dump_class]).merge(:name => property.field)
```

Comment:
Want to pass `property` directly to `type_by_property_class`? Then you can check `type_map` for both `class` and `dump_class`.